src/constraint/spec.ts

```diff
@@ -323,7 +324,14 @@ export const SPEC_CONSTRAINTS: SpecConstraintModel[] = [
     if (mark === Mark.AREA || mark === Mark.BAR) {
       for (let encQ of encodings) {
         if((encQ.channel === Channel.X || encQ.channel === Channel.Y) && encQ.scale) {
-          if (((encQ.scale as ScaleQuery).type as ScaleType) === ScaleType.LOG) {
+
+          const scale: ScaleQuery = encQ.scale as ScaleQuery;
+          let sType = scale.type as ScaleType;
+          let timeUnit = encQ.timeUnit as TimeUnit;
+          let type = encQ.type as Type;
+          let scaleType = _scaleType(sType, timeUnit, type);
```

Comment:
Umm, please read my comments carefully.   I mentioned that "use `_scaleType` as local variable and keep method name `scaleType`.

(That said, `sType` maybe better than `_scaleType`.)